demos/viscoelastic/viscoelastic_loading.py

```diff
@@ -0,0 +1,517 @@
+#
+# For the top surface we need to specify a normal stress, i.e. the weight of the ice load, as well as indicating this is a free surface.
+#
+# The `delta_rho_fs` option accounts for the density contrast across the free surface whether there is ice or air (or in later examples ocean!) above a particular region of the mantle.
+
+# +
+# Setup boundary conditions
+exterior_density = conditional(time < T2_load, rho_ice*disc, 0)
+stokes_bcs = {
+    bottom_id: {'uy': 0},
+    top_id: {'normal_stress': ice_load, 'free_surface': {'delta_rho_fs': density - exterior_density}},
+    1: {'ux': 0},
+    2: {'ux': 0},
+}
+
+gd = GeodynamicalDiagnostics(z, density, bottom_id, top_id)
+# -
+
+
+# We also need to specify a G-ADOPT approximation which sets up the various parameters and fields needed for the viscoelastic loading problem.
+
+
+approximation = SmallDisplacementViscoelasticApproximation(density, shear_modulus, viscosity, g=g)
+
+# We finally come to solving the variational problem, with solver
+# objects for the Stokes systems created. We pass in the solution fields `z` and various fields needed for the solve along with the approximation, timestep and boundary conditions.
+#
+
+stokes_solver = ViscoelasticStokesSolver(z, stress_old, displacement, approximation,
+                                         dt, bcs=stokes_bcs)
+
+# We next set up our output, in VTK format. This format can be read by programs like pyvista and Paraview.
+
+# +
+prefactor_prestress = Function(W, name='prefactor prestress').interpolate(approximation.prefactor_prestress(dt))
+effective_viscosity = Function(W, name='effective viscosity').interpolate(approximation.effective_viscosity(dt))
+
+if do_write:
+    # Create output file
+    output_file = VTKFile("output.pvd")
+    output_file.write(u_, displacement, p_, stress_old, shear_modulus, viscosity, density, prefactor_prestress, effective_viscosity, ice_load)
+
+plog = ParameterLog("params.log", mesh)
+plog.log_str(
+    "timestep time dt u_rms u_rms_surf ux_max disp_min disp_max"
+)
+
+checkpoint_filename = "viscoelastic_loading-chk.h5"
+# -
+
+# Now let's run the simulation! We are going to control the ice thickness using the `ramp` parameter. At each step we call `solve` to calculate the incremental displacement and pressure fields. This will update the displacement at the surface and stress values accounting for the time dependent Maxwell consitutive equation.
+
+for timestep in range(max_timesteps):
+    ramp.assign(conditional(time < T1_load, time / T1_load,
+                            conditional(time < T2_load, 1 - (time - T1_load) / (T2_load - T1_load),
+                                        0)
+                            )
+                )
+
+    ice_load.interpolate(ramp * rho_ice * g * Hice * disc)
```

Comment:
Whilst running this demo on `update_approx`, I realised that diagnostics change if `ice_load` is defined as some [UFL algebra](https://github.com/g-adopt/g-adopt/blob/63a38d0d9c1c3183a9dd6c3bdc309c46200de638/demos/viscoelastic/viscoelastic_loading.py#L404) instead of a `Function`. Changes are small enough to preserve the physical interpretation of the simulation but large enough to be [caught by the CI](https://github.com/g-adopt/g-adopt/actions/runs/11339720680).

Reply:
maybe it is the disc? I guess if we interpolate then we are forcing it to be a P1 field but otherwise it might do a different (better?) approximation of the tanh function if it is still an expression... @angus-g any ideas?

Reply:
So i've gone with your suggestion of doing the ice load symbolically as it is very simple. we are not currently plotting the ice load as well so I think this is fine. 

As discussed on zulip with a coarse mesh you can get fairly different 10% effects integrating the ice load over the surface whether using an expression (with default quadrature of 6) vs a P1 function. I think the expression is actually a bit more accurate (closer to the resolved value) for coarse meshes which makes sense.